src/tools/tidy/src/deps.rs

```diff
@@ -537,7 +535,7 @@ const PERMITTED_RUSTC_DEPENDENCIES: &[&str] = &[
     "windows_x86_64_gnu",
     "windows_x86_64_gnullvm",
     "windows_x86_64_msvc",
-    "wit-bindgen-rt@0.39.0", // pinned to a specific version due to using a binary blob: <https://github.com/rust-lang/rust/pull/136395#issuecomment-2692769062>
+    "wit-bindgen@0.46.0", // pinned to a specific version due to using a binary blob: <https://github.com/rust-lang/rust/pull/136395#issuecomment-2692769062>
```

Comment:
`wit-bindgen@0.46.0` contains a binary blob. It seems like we accidentally already got a dependency on this blob for the standard library again though.

Reply:
Ah yes the binary blob there is expected, and I'm happy to provide context for it if necessary. (unsure if you were looking for any other sort of particular feedback here)

Reply:
My understanding is that we intended to pin wit-bindgen-rt to a version from before the binary blob was introduced because some distros don't like binary blobs in source tarballs. I assume the reason you are using a binary blob is because of weak symbols. Would something like the binary blobs into a separate crate and then replacing this crate on the rustc side with `[patch.crates-io]` with a pure rust implementation that uses the unstable `#[linkage = "weak"]` or something like that work?